src/views/TechnicalAssistance/RequestForm/DocumentsTable.tsx

```diff
@@ -129,6 +144,40 @@ function DocumentsTable({
   ]);
 
   const columns = useMemo<Column<TrbRequestDocuments>[]>(() => {
+    const showViewError = () => {
+      showMessage(
+        <Alert type="error" className="margin-top-3">
+          {t('documents.viewFail')}
+        </Alert>
+      );
+    };
+
+    const getUrlForDocument = (documentId: string) => {
+      // use as promise so it's easier to download the file when getDocumentUrls() returns;
+      // this is a simpler approach instead of trying to do something fancy with the result status from useLazyQuery() (the called, loading, error, data booleans)
+      getDocumentUrls()
+        .then(response => {
+          if (response.error) {
+            showViewError();
+          } else if (response.loading) {
```

Comment:
IIRC Apollo propagates both GQL and network errors into the catch block/can be handled all in one place

Just a note of something Adam and I chatted about recently

Reply:
I just tested locally, looks like it's the other way around by default; the Promise returned from calling the query always gets fulfilled regardless of whether it's a GQL error or a network error, so a `.catch()` block shouldn't be needed. https://www.apollographql.com/docs/react/data/error-handling/#network-errors implicitly says this:
> When a network error occurs, Apollo Client adds it to [the error.networkError field returned by your useQuery call](https://www.apollographql.com/docs/react/data/queries/#result) (or whichever operation hook you used).

Reply:
Updated in https://github.com/CMSgov/easi-app/pull/2095/commits/dadda3a2db9a1cbccb406bef752b80e745c48342 to condense the error handling; now there's a single error handling block in `.then()` triggered by `if (response.error || !response.data)`, and no `.catch()`.

Reply:
> IIRC Apollo propagates both GQL and network errors into the catch block/can be handled all in one place

In light of the docs Dylan is pointing out, the error catch block we were using at the time might have been in play from a thrown error from form handling as part the mix in a form validation -> send mutation chain.

Appreciate the clarity on apollo. Just throwing up some of what I remember if there was confusion if I'm not mistaken.